pkg/azurestore/azureservice.go

```diff
@@ -67,87 +72,92 @@ type AzBlob interface {
 }
 
 type BlockBlob struct {
-	Blob       *azblob.BlockBlobURL
-	AccessTier azblob.AccessTierType
-	Indexes    []int
+	Blob           *blockblob.Client
```

Comment:
I'm a bit surprised by the field being named `Blob` but the type indicating it's a client object. Can you shed some light on this?

Reply:
yeah, that's ugly.. I tried to hard to not change too much :)
refactored this too